### PR TITLE
[with-typescript] Improved intial props examples, added example detail

### PR DESCRIPTION
**Pull request by resir014**
I've just thought of a way to improve the initial props page by adding
an example for a list/detail page structure. To do that, I've created a
separate `/detail` page, and a mock API which calls data from the array
we made on the previous PR.

A ListDetail component is created as an example for displaying detail.

Page structure is also cleaned up. Should I go ahead and add an example
on how to style with styled-jsx + its TS declarations? I might decide to
do it within this week anyway.

### CODE REVIEW

File: examples/with-typescript/pages/detail.tsx [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    }
+  }
+
+  render() {
+    const { item, errors } = this.props;
+
+    if (errors) {
+      return (
+        <Layout title={`Error | Next.js + TypeScript Example`}>
+          <p><span style={{ color: 'red' }}>Error:</span> {errors}</p>
```
Comment (resir014) [inline examples/with-typescript/pages/detail.tsx]: Not sure if I should show an example to extend the default Error component or not. 🤔

**Comment by resir014** [comment]
Just synced this PR with upstream.

File: examples/with-typescript/components/ListItem.tsx [hunk 2]
```diff
@@ -1,12 +1,16 @@
...
+
 import IDataObject from '../interfaces'
 
 type Props = {
   data: IDataObject,
 }
 
 const ListItem: React.FunctionComponent<Props> = ({ data }) => (
-  <React.Fragment>{data.id}:{data.name}</React.Fragment>
+  <Link href={`/detail?id=${data.id}`} passHref>
```
Comment (timneutkens) [inline examples/with-typescript/components/ListItem.tsx]: passHref is not required on Link tags, or is it when using typescript?

**Review by timneutkens — APPROVED** [review]
Looks alright 👍

File: examples/with-typescript/components/ListItem.tsx (hunk 2 above)
Comment (resir014) [reply examples/with-typescript/components/ListItem.tsx]: It should be fine without `passHref`, I'm just used to putting that in for `Link`s. 😅

**Pull request merged**